docker-compose/docker-compose.yml

```diff
@@ -1,10 +1,19 @@
-version: "3"
+# This docker-compose file will spin up an ACS cluster on a local host or on a server and it requires a minimum of 16GB Memory to distribute among containers.
+# Limit container memory and assign X percentage to JVM.  There are couple of ways to allocate JVM Memory for ACS Containers
+# For example: 'JAVA_OPTS: "$JAVA_OPTS -XX:+PrintFlagsFinal -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap"'
+# But, as per Oracle docs (https://docs.oracle.com/javase/9/gctuning/parallel-collector1.htm#JSGCT-GUID-CAB83393-3438-44ED-98F0-D15641B43C7D)
+# If container memory is not explicitly set, then the above flags will default max heap to 1/4th of container's memory which may not be ideal. 
+# Hence, setting up explicit Container memory and then assigning a percentage of it to the JVM for performance tuning.
+
+# Using version 2 as 3 does not support resource constraint options (cpu_*, mem_* limits) for non swarm mode in Compose  
```

Comment:
Do we no why this limitation was introduced (causing us to downgrade) ? Will this limit our ability to upgrade &/or make use of future docker-compose options ?

Reply:
Ref: https://docs.docker.com/compose/compose-file/#resources
```
Looking for options to set resources on non swarm mode containers?

The options described here are specific to the deploy key and swarm mode. If you want to set resource constraints on non swarm deployments, use Compose file format version 2 CPU, memory, and other resource options. If you have further questions, refer to the discussion on the GitHub issue docker/compose/4513.
```
As per the documentation and while using Version `3`, we were unable to set resource limits for non-swarm mode containers (i.e. Docker compose).

Reply:
Am wondering if it's really a problem going with v2 - it works with latest docker releases, v1 is still accepted. We're not aiming for Docker swarm.